lab-taylor-melanie-ken/lib/transform.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
```

Comment:
Seems like there is a handful of repetition in these functions. Most rely on the same looping construct. Is there a way to capture that functionality for reuse rather than retyping it each time?